src/builders.ts

```diff
@@ -851,3 +889,92 @@ export async function callReadOnlyFunction(
 
   return response.json().then(responseJson => parseReadOnlyResponse(responseJson));
 }
+
+/**
+ * Sponsored transaction options
+ *
+ * @param  {StacksTransaction} transaction - the origin-signed transaction to sponsor
+ * @param  {String} sponsorPrivateKey - the sponsor's private key
+ * @param  {BigNum} fee - the transaction fee amount to sponsor
+ * @param  {BigNum} sponsorNonce - the nonce of the sponsor account
+ * @param  {AddressHashMode} sponsorAddressHashmode - the sponsor address hashmode
+ */
+export interface SponsorOptions {
+  transaction: StacksTransaction;
+  sponsorPrivateKey: string;
+  fee?: BigNum;
+  sponsorNonce?: BigNum;
+  sponsorAddressHashmode?: AddressHashMode;
+}
+
+/**
+ * Constructs and signs a sponsored transaction as the sponsor
+ *
+ * @param  {SponsorOptions} sponsorOptions - the sponsor options object
+ *
+ * Returns a signed sponsored transaction.
+ *
+ * @return {ClarityValue}
+ */
+export async function sponsorTransaction(
+  sponsorOptions: SponsorOptions
```

Comment:
This is missing the `network?: StacksNetwork` argument

Reply:
@yknl I pushed a fix for this to this PR, hope you don't mind

Reply:
thanks for the fix!